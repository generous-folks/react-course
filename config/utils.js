const getExerciseFromProcess = () => {
  const exNumber = process.argv[2];
  if(!process.env.APP_EXERCISE || !exNumber) {
    return null;
  }

  return exNumber;
}

module.exports = getExerciseFromProcess;
