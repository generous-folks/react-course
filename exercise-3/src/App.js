import React from 'react';
import { Switch, Route } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';

export default function App() {
  return (
    <Switch>
      <Route path="/about">
        <AboutPage />
      </Route>
      <Route exact path="/">
        <HomePage />
      </Route>
    </Switch>
  );
}
