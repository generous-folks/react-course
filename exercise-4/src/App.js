import React from 'react';
import { Switch, Route } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';
import { ContactPage } from './pages/contact.page';

export default function App() {
  return (
    <Switch>
      <Route path="/about">
        <AboutPage />
      </Route>
      <Route path="/contact">
        <ContactPage />
      </Route>
      <Route exact path="/">
        <HomePage />
      </Route>
    </Switch>
  );
}
