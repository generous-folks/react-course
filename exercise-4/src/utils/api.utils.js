const endpoint = '/fixtures/articles.json';

export const getArticles = () =>
  fetch(endpoint)
    .then(res => res.json())
    .catch(err => console.log(err));
