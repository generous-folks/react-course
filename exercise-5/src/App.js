import React from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { HomePage } from './pages/home.page';
import { AboutPage } from './pages/about.page';
import { ContactPage } from './pages/contact.page';

export default function App() {
  return (
    <Router>
      <Switch>
        <Route path="/about">
          <AboutPage />
        </Route>
        <Route path="/contact">
          <ContactPage />
        </Route>
        <Route exact path="/">
          <HomePage />
        </Route>
      </Switch>
    </Router>
  );
}
