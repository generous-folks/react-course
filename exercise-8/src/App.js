import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { UserProvider } from './modules/user/user.context';

import { AppRoutes } from './modules/routing/components/routes.component';

export default function App() {
  return (
    <UserProvider>
      <Router>
        <AppRoutes />
      </Router>
    </UserProvider>
  );
}
