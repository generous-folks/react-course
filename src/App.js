import React from "react";
import { Switch, Route } from "react-router-dom";

import { Home } from "./pages/home.page";

export default function App() {
  return (
        <Switch>
          <Route path="/about">
            {/* <About /> */}
          </Route>
          <Route path="/articles/:id">

          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
  );
}
