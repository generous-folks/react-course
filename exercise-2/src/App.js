import React, { useState, useEffect } from 'react';
import { getArticles } from './utils/api.utils';

export default function App() {
  const [articles, setArticles] = useState([]);

  useEffect(() => {
    try {
      if (articles.length === 0) {
        getArticles()
          .then(res => setArticles(res))
          .catch(err => console.error(err));
      }
    } catch (error) {
      throw new Error(`Fetch articles failed: ${error.message}`);
    }
  }, [articles]);

  return (
    <div>
      <h2>Home Page</h2>
      <div>
        <h4>Articles</h4>
        <ul>{articles.length > 0 && articles.map(({ id, name }) => <li key={id}>{name}</li>)}</ul>
      </div>
    </div>
  );
}
