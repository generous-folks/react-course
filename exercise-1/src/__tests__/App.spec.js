import React, { useEffect, useState } from 'react';

import { shallow } from 'enzyme';

import App from '../App';

import MockedArticles from '../../../fixtures/articles.json';

import * as ApiUtils from '../utils/api.utils';

let wrapper;
const emptyArray = [];

ApiUtils.getArticles = jest.fn().mockResolvedValue(MockedArticles);

jest.mock('react', () => global.mockReactWithHooks({ effect: true, state: true }));

const getWrapper = () => shallow(<App />);

beforeEach(() => {
  jest.clearAllMocks();
  useEffect.mockClear();
  useState.mockClear();
  wrapper = getWrapper();
});

xdescribe('App', () => {
  describe('State checks', () => {
    // Don't test Jest mocks or React, it is useless
    // This tests that our react mock is working, useless
    // const [initialsArticles] = useState.mock.results[0].value;
    // expect(initialsArticles).toEqual(emptyArray);

    it('should call useState with an empty array', () => {
      expect(useState).toHaveBeenCalledWith(emptyArray);
    });
  });

  describe('Effects checks', () => {
    it('should call useEffect with a function and an empty array', () => {
      expect(useEffect).toHaveBeenCalledWith(expect.any(Function), [emptyArray]);
    });

    it('should call getArticles once', () => {
      expect(ApiUtils.getArticles).toHaveBeenCalledTimes(1);
    });

    it('should call getArticles once even when re-rendered', () => {
      wrapper.update();
      expect(ApiUtils.getArticles).toHaveBeenCalledTimes(1);
    });

    it('should call setArticles once with MockedArticles', () => {
      const [, setArticles] = useState.mock.results[0].value;
      expect(setArticles).toHaveBeenCalledWith(MockedArticles);
      expect(setArticles).toHaveBeenCalledTimes(1);
    });
  });
});
