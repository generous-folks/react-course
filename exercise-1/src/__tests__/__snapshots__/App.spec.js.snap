// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <h2>
    Home Page
  </h2>
</div>
`;
